core/txpool/validation.go

```diff
@@ -207,6 +210,23 @@ func ValidateTransactionWithState(tx *types.Transaction, signer types.Signer, op
 	if next > tx.Nonce() {
 		return fmt.Errorf("%w: next nonce %v, tx nonce %v", core.ErrNonceTooLow, next, tx.Nonce())
 	}
+	// Ensure the transaction is allowed by policy
+	// Suppose the storage of Policy contract is like
+	// contract Policy {
+	//     uint256 minGasPrice;
+	//     mapping(address=>bool) blacklist;
+	// }
+	// then minGasPrice is slot 0, blacklist is slot 1
+	// Apply policy minimum gas price
+	var minGasPrice = opts.State.GetState(opts.PolicyAddress, common.BytesToHash([]byte{0}))
+	if tx.GasPrice().Cmp(minGasPrice.Big()) < 0 {
+		return fmt.Errorf("%w: policy needed %v, tx permitted %v", ErrUnderpriced, minGasPrice.Big(), tx.GasPrice())
+	}
+	// Apply policy blacklist
+	var blocked = opts.State.GetState(opts.PolicyAddress, crypto.Keccak256Hash(common.LeftPadBytes(from.Bytes(), 32), common.LeftPadBytes([]byte{1}, 32)))
+	if blocked != (common.Hash{}) {
+		return fmt.Errorf("%w: sender %s", ErrBlockedSender, from.Hex())
+	}
```

Comment:
```suggestion
	// then minGasPrice is slot 1, blacklist is slot 2
	// Apply policy minimum gas price
	var minGasPrice = opts.State.GetState(opts.PolicyAddress, common.BytesToHash([]byte{1}))
	if tx.GasPrice().Cmp(minGasPrice.Big()) < 0 {
		return fmt.Errorf("%w: policy needed %v, tx permitted %v", ErrUnderpriced, minGasPrice.Big(), tx.GasPrice())
	}
	// Apply policy blacklist
	var blocked = opts.State.GetState(opts.PolicyAddress, crypto.Keccak256Hash(common.LeftPadBytes(from.Bytes(), 32), common.LeftPadBytes([]byte{2}, 32)))
	if blocked != (common.Hash{}) {
		return fmt.Errorf("%w: sender %s", ErrBlockedSender, from.Hex())
	}
```
`Policy` contract inherits from `GovernanceVote` contract, slot 0 is `voteMap`.